service/src/main/resources/api/openapi.yml

```diff
@@ -742,6 +754,93 @@ paths:
 components:
   schemas:
     # Please keep alphabetized
+    ClusterMetadata:
+      type: object
+      required:
+      - configBucket
+      - tempBucket
+      - managerNodeConfig
+      - primaryWorkerConfig
+      properties:
+        initializationActions:
+          type: array
+          items:
+            type: string
+          description: Initialization actions for the cluster.
+        configBucket:
+          type: string
+          description: The config bucket name used by the cluster
+        tempBucket:
+          type: string
+          description: The temp bucket name used by the cluster
+        autoscalingPolicy:
+          type: string
+          description: URI of autoscaling policy.
+        metadata:
+          type: object
+          description: Metadata key-value pairs.
+          additionalProperties:
+            type: string
+        managerNodeConfig:
+          $ref: '#/components/schemas/ClusterInstanceGroupConfig'
+        primaryWorkerConfig:
+          $ref: '#/components/schemas/ClusterInstanceGroupConfig'
+        secondaryWorkerConfig:
+          $ref: '#/components/schemas/ClusterInstanceGroupConfig'
+        lifecycleConfig:
+          $ref: '#/components/schemas/ClusterLifecycleConfig'
+
+    ClusterInstanceGroupConfig:
+      type: object
+      required:
+      - numInstances
+      - machineType
+      properties:
+        numInstances:
+          type: integer
+          format: int32
+          description: Number of instances in the instance group.
+        machineType:
+          type: string
+          description: Compute engine machine type.
+        imageUri:
+          type: string
+          description: Compute engine image used for cluster instances.
+        bootDiskSizeGb:
+          type: integer
+          format: int32
+          default: 500
+          description: The size of the boot disk in GB.
+        preemptibility:
+          type: string
+          enum: [ "NON_PREEMPTIBLE", "PREEMPTIBLE", "SPOT" ]
+          description: Instance preemption behavior.
+
+    ClusterLifecycleConfig:
+      description: >-
+        The lifecycle options for a cluster.
+        See https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig#LifecycleConfig
+      type: object
+      properties:
+        idleDeleteTtl:
+          type: string
+          description: Duration of idling before cluster is deleted. See https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/scheduled-deletion.
+          example: 1800s
+        idleStartTime:
+          type: string
+          format: date-time
+          description: Time when cluster became idle.
+          example: 2014-10-02T15:01:23.045123456Z
+        autoDeleteTime:
```

Comment:
fyi - https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/
this might not be as straightforward here, since there are other properties.

Reply:
Going to drop this constraint. It matters more when setting these values (in WSM). Only one of these will exist when fetched from gcloud.